### PR TITLE
Replace passing by const std::unique_ptr<T>& by const T&

### PR DESCRIPTION
**Pull request by masterleinad**
https://github.com/dealii/dealii/pull/11698#issuecomment-774854191:
> It is correct that in pattern.h, we could replace the std::unique_ptr arguments for the to_string() and to_value() functions to regular const references. I would accept a patch to this end.

This is what this pull request does. I am in favor of doing this but it's not backward-compatible. Let's see what people think.

### CODE REVIEW

File: include/deal.II/base/parameter_handler.h [hunk 1]
```diff
@@ -2334,7 +2334,7 @@ ParameterHandler::add_parameter(const std::string &          entry,
 
   declare_entry(entry,
                 Patterns::Tools::Convert<ParameterType>::to_string(
-                  parameter, pattern.clone()),
+                  parameter, *pattern.clone()),
```
Comment (bangerth) [inline include/deal.II/base/parameter_handler.h:2337]: Here and elsewhere, you don't actually need the `clone()` call any more.
```suggestion
                  parameter, pattern),
```

**Review by bangerth — COMMENTED** [review]
Let's let @luca-heltai make the call.

**Comment by bangerth** [comment]
@luca-heltai Ping?

**Comment by luca-heltai** [comment]
Ah, sorry. I totally missed this one. 

This is technically an incompatible change. 95% of my codes will break, since I often use the feature of passing a custom pattern. However, I'm in favour of this change. I don't remember why I did it this way in the first place. I suspect that something will break, but I cannot foresee what will....

**Review by luca-heltai — APPROVED** [review]

**Comment by masterleinad** [comment]
I added a changelog entry. Ready to be merged from my side.

**Pull request merged**